### PR TITLE
Add missing currencyDisplay to resolved number format options

### LINKED ISSUE #43958: ResolvedNumberFormatOptions is missing currencySign option
# Bug Report

currencySign is missing from NumberFormatOptions

### 🔎 Search Terms

currencySign

### 🕗 Version & Regression Information

I can see it's missing in the types for es5, just as it was added to the options here https://github.com/microsoft/TypeScript/pull/40709 I'll create a pull tomorrow hopefully

### ⏯ Playground Link

[Updated Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5IBQANkgC6gBmc+AtgIbnkGgC8oAdkgO6gCSnciQB0AOQCudAEYEAYrUbkAFAHIknALRiAWqoA0oAN6hM5SGQBcoVTAn5CnGJAOx7j59dW6AIq7sOGs4AygCWqJxeDDDwEkKhnKiqoAC+AJQA3EQ09Ews+CKEmHAkAG5ICADyAA7koXCcmMppIgEekGERWUA)

### 💻 Code

```ts
let formatter = new Intl.NumberFormat('en-NZ', { style: 'currency', currency: 'NZD', currencySign: 'accounting' });
formatter.resolvedOptions().currencySign;
```

### 🙁 Actual behavior

I get a TS error `Property 'currencySign' does not exist on type 'ResolvedNumberFormatOptions'.(2339)`

### 🙂 Expected behavior

I shouldn't get a TS error

### PR DESCRIPTION
**Pull request by snowystinger**
Please verify that:
* [x] There is an associated issue in the `Backlog` milestone (**required**)
* [x] Code is up-to-date with the `master` branch
* [x] You've successfully run `gulp runtests` locally
* [x] There are new or updated unit tests validating the change

Refer to CONTRIBUTING.MD for more details.
  https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md

Fixes #43958

### CODE REVIEW

**Comment by almeidx** [comment]
> I'm not sure why it thinks I don't have a linked issue, it's in the description. Probably didn't format that correctly.

I believe it expects you to just include the `#number` (`Fixes #43958`) of the pull request, but you included the whole URL.

File: src/lib/es5.d.ts [hunk 1]
```diff
@@ -4311,6 +4311,7 @@ declare namespace Intl {
         style: string;
         currency?: string;
         currencyDisplay?: string;
+        currencySign?: string;
```
Comment (CreativeTechGuy) [inline src/lib/es5.d.ts:4314]: Shouldn't this, and `currencyDisplay`, go in the [es2020.intl.d.ts](https://github.com/microsoft/TypeScript/blob/master/src/lib/es2020.intl.d.ts#L264-L278) interface instead?

From MDN [Compatibility Chart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#browser_compatibility) it looks like this was released at the same time as all of the other options listed in the es2020 interface.

Comment (sandersn) [reply src/lib/es5.d.ts]: @DanielRosenwasser [previously commented](https://github.com/microsoft/TypeScript/pull/40876#issuecomment-702463480) that Intl isn't part of the ecmascript standard, so I think it should actually stay in ES5. Daniel, can you confirm?

Comment (orta) [reply src/lib/es5.d.ts]: We want this PR as it is, es2020 is the right pattern - these APIs may not tie directly to the esXXXX specs but they are released and supported in a similar way to how we structure these libs. This came out in 2020 and we should reflect that via the `target` - which this PR does.

**Review by orta — APPROVED** [review]
This PR is good and we should get it in for 4.5 - I'll give Daniel a few days to give any last opinions and I'll merge otherwise.

**Comment by orta** [comment]
I've rebased this after https://github.com/microsoft/TypeScript/pull/45647 got merged and assuming it's green, I'll merge tomorrow

**Comment by sandersn** [comment]
Looks like the new tests are still failing after the merge from main.

**Comment by snowystinger** [comment]
gentle nudge about the tests from main

**Comment by sandersn** [comment]
Once the tests pass I'll merge this for 4.6
Thanks @DanielRosenwasser for remembering this.

**Review by sandersn — APPROVED** [review]

**Pull request merged**